### ISSUE TITLE
Drop the requirement of no locked platforms in pathways.txt

### ISSUE THREAD

**Issue by miklcct** [google/transit#517]
### Describe the problem

The specifications of `pathways.txt` mentions that,

> No locked platforms: Each platform (location_type=0 or empty) or boarding area (location_type=4) must be connected to at least one entrance/exit (location_type=2) via some chain of pathways. Stations not allowing a pathway to the outside of the station from a given platform are rare.

The fact is that, there exist some stations where access to the outside world is prohibited. These stations are for interchange only. The most famous example in Hong Kong is called Shing Mun Tunnels Bus Interchange, which is an interchange-only station with no public access to outside the station.

### Use cases

We need to define stations which are interchange only, i.e. no access to the outside world.

### Proposed solution

Delete the quoted paragraph. Change the validator to produce a warning instead.

**Comment by gcamp** [comment]
I'm afraid of doing this because it's the error that we see the most often in pathways. Relaxing the requirement will just make things worst. 

Even if the example that you provided is valid, the benefit of pathways in Shing Mun Tunnels Bus Interchange is marginal. What is your expectations, just walkways between platforms?

**Comment by miklcct** [comment]
> I'm afraid of doing this because it's the error that we see the most often in pathways. Relaxing the requirement will just make things worst.
> 
> Even if the example that you provided is valid, the benefit of pathways in Shing Mun Tunnels Bus Interchange is marginal. What is your expectations, just walkways between platforms?

Yes, walkway between platforms. The same situation happened in Prince Edward MTR station between 10 May 1982 and 16 May 1982, when the station was still under construction.
